### PR TITLE
Recommend buildSrc over script plugins

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

As discussed recommend `buildSrc` over script plugins and add more details.

### CODE REVIEW

**Comment by eskatos** [comment]
`buildSrc` is a (included) *build* not a *project*. I wonder if calling it a *project* could be misleading for users. It is to me. Getting the terminology straight now might help when we start recommending included builds.

**Comment by bmuschko** [comment]
@eskatos I know that we have been talking about treating `buildSrc` as part of a composite build in the past but has this actually become reality yet?

> I wonder if calling it a project could be misleading for users. It is to me.

Can you explain why you find it misleading? At least from a user's perspective you see that another _project_ with the path `:buildSrc` is executed. The console output simply looks similar to what you'd expect in a multi-project build.

**Comment by eskatos** [comment]
Both included builds and sub-projects are displayed in a similar way in the console output, respectively `":${includedBuild.rootProject.name}:task"` or `":${subproject.name}:task"`. This might be the misleading part.

`buildSrc` is already a "special included build", not a "special project". It can have a settings file for example.

**Comment by bmuschko** [comment]
@eskatos Thanks for clarifying. I will change it.

File: subprojects/docs/src/docs/userguide/organizingGradleProjects.adoc [hunk 1]
```diff
@@ -145,19 +146,26 @@ It also provides direct access to the Gradle API. Additional dependencies can be
 ++++
 
 A typical project including `buildSrc` has the following layout.
+Any code under `buildSrc` should use a package similar to application code.
```
Comment (oehme) [inline subprojects/docs/src/docs/userguide/organizingGradleProjects.adoc]: This line is confusing. Does it mean the code under buildSrc should have a similar package name as your application code?

I know it doesn't but that's what it reads like. I'd remove this sentence. buildSrc is a java project like any other, so all the same best practices apply.

**Pull request merged**